utils/fd-metrics/src/lib.rs

```diff
@@ -69,6 +69,7 @@ pub fn fd_metrics() -> HashMap<FdType, usize> {
 /// Maps fd to FdType
 ///
 /// Uses fstat on unix, always returns None on Windows
+#[allow(unused_variables)]
```

Comment:
`#[cfg_attr(target_os = "windows", unused_variables)]` would be more precise, but I'm not sure if it's not too much clutter. I have no strong preference, so feel free to merge if you want to keep it that way.